consul/map.jinja

```diff
@@ -13,10 +15,9 @@
         }
   }
   ,grain="cpuarch"
-  ,merge=salt['pillar.get']('consul'))
+  ,merge=consul)
 %}
-{% do defaults.consul.update(cpu_arch_map) %}
 
-{% set consul = salt['pillar.get']('consul', default=defaults.consul, merge=True) %}
+{% do consul.update(cpu_arch_map) %}
```

Comment:
Hey @vutny. it was actually me who added the cpu_arch_map so we could fetch the appropriate Hashicorp package depending on the architecture of the machine we'd be deploying consul to, so now you know what it was meant to be. And your fix looks actually much better than what I did there, so I will probably migrate it to the other packages using the same strategy. :)